chore(deps): upgrade to react 17

dimitropoulos opened the pull request:
In a year and a half since release of react 17, it's been more and more common that we hit blockers in upgrading other packages because our react version is so far behind (and unsupported by the new version of the tool we're trying to use).

This PR tries to do the bare minimum to upgrade react.  Everything is working fairly well, except one dependency `framer-motion`.  I bisected and determined that `framer-motion@5.0.0` is the exact version where the break occurs.  Thankfully, the version before that also supports react 17, which means we can defer whatever the problem is with `framer-motion` for now.  To whoever deals with this problem with `framer-motion` in the future, start [here](https://github.com/framer/motion/issues/1307#issuecomment-955877279).

Here's the output of `npx check-peer-dependencies --npm`

| BEFORE | AFTER |
| - | - |
| ![Screenshot_20220224_163656](https://user-images.githubusercontent.com/15232461/155616723-25e0cd77-bb8a-4de3-b87c-a2c03d26bb1c.png) | no mismatches |

dimitropoulos committed:
removes (now not erroring) expect-error directive

jackkav commented:
cool if we delay this until after #4529 to keep the variables to a minimum?

gatzjames reviewed (approved):
Tested and works fine 🚀
Not sure what the effect is with what @jackkav said (https://github.com/Kong/insomnia/pull/4521#issuecomment-1053541181) but we should make sure it doesn't affect that PR before merging.

dimitropoulos commented:
(update, we're going to hold off on this until after a GA release with electron goes out)

dimitropoulos committed:
upgrades react

dimitropoulos committed:
upgrades react-switch to support react 17

https://github.com/markusenglund/react-switch/blob/master/CHANGELOG.md

dimitropoulos committed:
upgrades react-sortable-hoc to support react 17

https://github.com/clauderic/react-sortable-hoc/blob/master/CHANGELOG.md

dimitropoulos committed:
upgrades react-tabs to support react 17

https://github.com/reactjs/react-tabs/releases

dimitropoulos committed:
upgrades swagger-ui-react to support react 17

https://github.com/swagger-api/swagger-ui/releases?page=2

dimitropoulos committed:
upgrades framer-motion to highest non-breaking version

I bisected and determined that the break for us occurs exactly at version 5.0.0.  Thankfully, the version before this also supports react 17.

dimitropoulos committed:
package locks
